### DISCUSSION TITLE
wire:loading for all instances of a component

### DISCUSSION THREAD

**Discussion by randohinn** [livewire/livewire#5072] [1. Help]
Hi all. 

I have a component that contains this piece of code

```blade
<div wire:loading.remove x-show="!open" class="inline-flex flex-col h-full w-6 bg-neutral-100">
                @unless($first)
                    <div wire:click="moveUp" class="h-1/2 flex items-center hover:bg-neutral-200">
                        <svg width="24" height="24" fill="none" viewBox="0 0 24 24">
                            <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M15.25 14.25L12 10.75L8.75 14.25"></path>
                        </svg>
                    </div>
                @endunless
                @unless($last)
                    <div wire:click="moveDown" class="h-1/2 flex items-center hover:bg-neutral-200">
                        <svg width="24" height="24" fill="none" viewBox="0 0 24 24">
                            <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M15.25 10.75L12 14.25L8.75 10.75"></path>
                        </svg>
                    </div>
                @endunless
            </div>
```

This component gets rendered x amount of times in a loop. `wire:click="moveUp"` and `wire:click="moveDown"` end up reshuffling the order they get rendered in. What I therefore want, is to hide all of these divs on every component, immediately on click. Currently, the one that was clicked hides immediately, and then afterwards, the others hide too. 

How could I achieve this in such a way that hiding would occur on every instance, immediately on click?

**Comment by Martleby** [comment]
How about dispatching a browser event from the component that you can listen for on each component?

PS I would recommend adding wire:key to the components in the loop if you haven't already.
